zedd-app/src/components/SettingsDialog.tsx

```diff
@@ -101,6 +101,36 @@ export const SettingsDialog = observer(
     const projects = uniq([...clarityState.projectNames, ...settings.excludeProjects])
     projects.sort()
 
+    function updateWorkmask(newValueofHours: any) {
+      let indexToChange = 0
+      let actualSumOfHours = settings.workmask.reduce(
+        (total, day) => (total = total + day.valueOf()),
+        0,
+      )
+      let workingDays = settings.workmask.slice(0, 5)
+      let valueToChange
+      if (newValueofHours - actualSumOfHours > 0) {
+        valueToChange = Math.min(...workingDays)
+      } else {
+        valueToChange = Math.max(...workingDays)
+      }
+      indexToChange = workingDays.indexOf(valueToChange)
+
+      let value = 1
+      if (newValueofHours < actualSumOfHours) {
+        value = -value
+      }
+      settings.workmask[indexToChange] = settings.workmask[indexToChange] + value
+    }
+
+    function changeDayHours(e: any): number {
```

Comment:
you can shorten the whole function even further by inlining the constant variables